ui/app/components/bn-as-decimal-input.js

```diff
@@ -0,0 +1,143 @@
+const Component = require('react').Component
+const h = require('react-hyperscript')
+const inherits = require('util').inherits
+const ethUtil = require('ethereumjs-util')
+const BN = ethUtil.BN
+const extend = require('xtend')
+
+module.exports = BnAsDecimalInput
+
+inherits(BnAsDecimalInput, Component)
+function BnAsDecimalInput () {
+  this.state = { invalid: null }
+  Component.call(this)
+}
+
+/* Bn as Decimal Input
+ *
+ * A component for allowing easy, decimal editing
+ * of a passed in bn string value.
+ *
+ * On change, calls back its `onChange` function parameter
+ * and passes it an updated bn string.
+ */
+
+BnAsDecimalInput.prototype.render = function () {
+  const props = this.props
+  const state = this.state
+
+  const { value, precision, onChange, min, max } = props
+
+  const suffix = props.suffix
+  const style = props.style
+  const scale = Math.pow(10, precision)
+  const newValue = value.toNumber(10) / scale
```

Comment:
On this line, you are assuming that the input number fits in a JS number, which defeats the point of using BN.

I've added a test to demonstrate the problem, which highlights why this doesn't work for editing ether values, for example.